ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -340,20 +347,39 @@ static void createCommandTopicIfNecessary(final KsqlRestConfig restConfig,
                 .toString()
         );
       }
-      if(replicationFactor < 2) {
-        log.warn("Creating topic %s with replication factor of %d which is less than 2. "
-            + "This is not advisable in a production environment. ", replicationFactor);
+      if (replicationFactor < 2) {
+        log.warn("Creating topic {} with replication factor of %d which is less than 2. "
+                 + "This is not advisable in a production environment. ", replicationFactor);
       }
 
       // for now we create the command topic with infinite retention so that we
       // don't lose any data in case of fail over etc.
       topicClient.createTopic(commandTopic,
-          1,
-          replicationFactor,
-          Collections.singletonMap(TopicConfig.RETENTION_MS_CONFIG,
-              String.valueOf(Long.MAX_VALUE)));
+                              1,
+                              replicationFactor,
+                              Collections.singletonMap(TopicConfig.RETENTION_MS_CONFIG,
+                                                       String.valueOf(Long.MAX_VALUE)));
     } catch (KafkaTopicException e) {
-      log.info("Command Topic Exists: " + e.getMessage());
+      log.info("Command Topic Exists: {}", e.getMessage());
+    }
+  }
+
+  private static void loadUiWar() {
+    log.info("Loading UI-WAR from {}", new File(UI_FOLDER).getAbsolutePath());
+    final File[] files = new File(UI_FOLDER)
+        .listFiles((dir, name) -> name.endsWith(".war"));
+
+    if (files != null) {
```

Comment:
What if `files == null` or `files.length != 1` ? I assume we expect exactly one war?
If the user has enabled the UI and there are no war files they will not be warned. If there is more than one, then which one are we using?

Reply:
Well, this is Neil's change, but I would of thought... 

`files == null` just means the user hasn't put the ui war file in the necessary directory. So this would be normal operation.

`files.length > 1`, well, then we'd just extract all the war files.